native/test/src/tests/msg_edit.rs

```diff
@@ -1,41 +1,85 @@
+use tracing::info;
 
-use crate::utils::random_user_with_random_space;
+use crate::utils::random_user_with_random_convo;
 
 #[tokio::test]
 async fn message_edit() -> Result<()> {
     let _ = env_logger::try_init();
 
-    let (mut user, room_id) = random_user_with_random_space("message_edit").await?;
+    let (mut user, room_id) = random_user_with_random_convo("message_edit").await?;
     let state_sync = user.start_sync();
     state_sync.await_has_synced_history().await?;
 
-    let space = user.space(room_id.to_string()).await?;
-    let event_id = space.send_plain_message("Hi, everyone".to_string()).await?;
+    let convo = user.convo(room_id.to_string()).await?;
+    let timeline = convo.timeline_stream().await?;
+    let stream = timeline.diff_stream();
+    pin_mut!(stream);
+
+    let event_id = convo.send_plain_message("Hi, everyone".to_string()).await?;
     println!("event id: {event_id:?}");
 
-    let edited_id = space
+    let edited_id = convo
         .edit_plain_message(event_id.to_string(), "This is message edition".to_string())
         .await?;
+    println!("edited id: {edited_id:?}");
 
-    let ev = space.event(&edited_id).await?;
-    println!("msg edition: {ev:?}");
-
-    let Ok(AnyTimelineEvent::MessageLike(AnyMessageLikeEvent::RoomMessage(MessageLikeEvent::Original(r)))) = ev.event.deserialize() else {
-        bail!("not the proper room event")
-    };
-
-    let msg = RoomMessage::room_message_from_event(r, space.deref().deref().clone(), false);
-    let Some(item) = msg.event_item() else {
-        bail!("This item should be event item not virtual item")
-    };
+    while let Some(diff) = stream.next().await {
+        match diff.action().as_str() {
+            "Append" => {}
+            "Insert" => {}
+            "Set" => {
+                info!("diff index: {:?}", diff.index());
+                info!("diff value: {:?}", diff.value());
+                info!("diff values: {:?}", diff.values());
+                let Some(value) = diff.value() else {
+                    bail!("diff set action should have valid value")
+                };
+                assert_eq!(value.item_type(), "event");
+                let Some(event_item) = value.event_item() else {
+                    bail!("user already sent message edition")
+                };
+                assert!(event_item.is_edited());
+                assert_eq!(event_item.event_id(), event_id.to_string()); // should be original id, because original msg is replaced with msg edition
+                let Some(text_desc) = event_item.text_desc() else {
+                    bail!("message edition should have text desc")
+                };
+                assert_eq!(text_desc.body(), "This is message edition");
+                break;
+            }
+            "Remove" => {}
+            "PushBack" => {}
+            "PushFront" => {}
+            "PopBack" => {}
+            "PopFront" => {}
+            "Clear" => {}
+            "Reset" => {
```

Comment:
I am not sure I understand ... we expect either Reset or set? Don't we surely know which one?

Reply:
We get `Hi, everyone` msg under `PushBack` action in usual case.
We may get that msg under `Reset` action, when calling of `now_or_never()` loop was too late and timeline already arranged `Hi, everyone` into stream.
We get `This is message edition` msg under `Set` action.
As we know, `Set` action means by record modification.